### PR TITLE
Add network topology APIs

### PR DESCRIPTION
**Pull request by zheng1**
Signed-off-by: Zhengyi Lai <zheng1@yunify.com>

**What type of PR is this?**
> /kind feature

**What this PR does / why we need it**:
Add network topology APIs

### CODE REVIEW

**Comment by zryfish** [comment]
Please submit to dev branch

File: pkg/apiserver/network/topology.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	if err != nil {
+		log.Printf("write response failed %v", err)
+	}
+}
+
+func GetNamespaceNodeTopology(request *restful.Request, response *restful.Response) {
+	var query = url.Values{
+		"namespace": []string{request.PathParameter("namespace")},
+		"timestamp": request.QueryParameters("timestamp"),
```
[the remark this answers is not in the document]
Comment (zheng1) [reply pkg/apiserver/network/topology.go:50]: If there is no timestamp, scope returns the latest data by default

**Comment by zheng1** [comment]
> Please submit to dev branch

Done

File: pkg/kapis/network/v1alpha2/handler.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
+package v1alpha2
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
```
Comment (zryfish) [inline pkg/kapis/network/v1alpha2/handler.go:6]: we are using klog instead of log

File: pkg/kapis/network/v1alpha2/handler.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	"net/http"
+	"net/url"
+
+	restful "github.com/emicklei/go-restful"
+)
+
+const ScopeQueryUrl = "http://weave-scope-app.weave.svc/api/topology/services"
```
Comment (zryfish) [inline pkg/kapis/network/v1alpha2/handler.go:13]: This means weave-scope is required for network topology, but how to enable and disable it? And we need to provide API for console to find out which components are enabled, check `pkg/apiserver/config`

File: pkg/kapis/network/v1alpha2/handler.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		"namespace": []string{request.PathParameter("namespace")},
+		"timestamp": request.QueryParameters("timestamp"),
+	}
+	var u = fmt.Sprintf("%s?%s", ScopeQueryUrl, query.Encode())
+
+	resp, err := http.Get(u)
+
+	if err != nil {
+		log.Printf("query scope faile with err %v", err)
+		_ = response.WriteError(http.StatusInternalServerError, err)
```
Comment (zryfish) [inline pkg/kapis/network/v1alpha2/handler.go:26]: use `api.HandleInternalError` instead

**Pull request merged**